src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/EnumGroupAndName.cs

```diff
@@ -39,6 +77,59 @@ public EnumGroupAndName(string group, string name)
         /// <summary>
         /// Gets the name.
         /// </summary>
-        public string Name { get; }
+        public string Name
+        {
+            get
+            {
+                return GetDisplayName();
+            }
+        }
+
+        // Equals and GetHashCode must be overloaded to accomidate the _nameFunc
+        public override bool Equals(object obj)
+        {
+            if (!(obj is EnumGroupAndName))
+            {
+                return false;
+            }
+
+            var second = (EnumGroupAndName)obj;
+
+            return string.Equals(Group, second.Group) && string.Equals(Name, second.Name);
+        }
+
+        public override int GetHashCode()
+        {
+            var hashcode = HashCodeCombiner.Start();
+
+            hashcode.Add(Group);
+            hashcode.Add(Name);
+
+            return hashcode;
+        }
+
+        private string GetDisplayName()
+        {
+            if (_fieldInfo == null)
+            {
+                return _name;
+            }
+            else
+            {
+                var display = _fieldInfo.GetCustomAttribute<DisplayAttribute>(inherit: false);
```

Comment:
This call is expensive and not something we need to do repeatedly. Add another field to store the `DisplayAttribute` and set `_name` to `_fieldInfo.Name` so that this is done at most once even when there's no such attribute.